### ISSUE TITLE
MonoGame only supports 4 RenderTargets, while DX10+ supports 8

### ISSUE THREAD

**Issue by RendijsSmukulis** [MonoGame/MonoGame#6367]
Calling `this.GraphicsDevice.SetRenderTargets` with more than 4 render targets throws an exception when the new render targets are copied to `_currentRenderTargetBindings`, [here](https://github.com/MonoGame/MonoGame/blob/develop/MonoGame.Framework/Graphics/GraphicsDevice.cs#L830).

This is caused by harcoded array length of 4 [here](https://github.com/MonoGame/MonoGame/blob/develop/MonoGame.Framework/Graphics/GraphicsDevice.DirectX.cs#L65) and [here](https://github.com/MonoGame/MonoGame/blob/develop/MonoGame.Framework/Graphics/GraphicsDevice.cs#L58).

I'd be happy to provide a test case and a pull request that ups these limits, unless there are reasons I'm not seeing to limit this at 4 (e.g. compatibility with platforms other than WindowsDX)? Otherwise, improving the exception thrown would also be great, currently we just get an array overflow.

#### What version of MonoGame does the bug occur on:
- MonoGame 3.6

#### What operating system are you using:
- Windows

#### What MonoGame platform are you using:

- WindowsDX

**Comment by MattiKloepper** [comment]
More than four rendertargets would suit me well, too.
Maybe some sort of "capabilities querying" could be implemented so that we will be able to use 4+ rendertargets on devices that support this and provide a fallback path for other Hardware?

**Comment by Jjagg** [comment]
Yes, I agree we should add max nr of RTs to GraphicsCapabilities. 

When #1941 is solved GraphicsCapabilities of a GraphicsDevice can be retrieved to check what exactly is supported under the set profile or by the gpu (in case no profile is set to limit capabilities).

**Comment by tomspilman** [comment]
@Jjagg - I wouldn't be opposed to bending the existing XNA rules to allow "at least 4 render targets" under `HiDef`.  This might be a good first step.

**Comment by MattiKloepper** [comment]
Travelled a bit through the source code and checked some Direct3D documentation.
The maximum number of simultaneous render targets seems to be defined just by the feature Level:

-    <= 9.2 => 1 Target
-    ==9.3 => 4 Targets
-    >= 10.0 => 8 Targets

Additionally the is a constant "SharpDX.Direct3D11.PixelShaderStage.OutputRegisterCount". Does anybody know if this constant refers to the max number of render targets?

Based on this I guess I could add a PR to complement the GraphicsCapabilities with a max nr of rendertargets property. I would propose to set that property to 4 in the Default case and to "OutputRegisterCount" if the DirectX feature Level is >= 10.0.

**Comment by piotrplata** [comment]
I've recently migrated from DX11 to DX12. 8 render targets worked fine on DX11 HiDef, but on DX12 it stops at 4.

I have a minimal DX12 repro here:
https://github.com/piotrplata/MonoGameBugReports/tree/main/Repros/02-Multiple-Render-Targets

The repro creates five offscreen render targets and throws when calling `GraphicsDevice.SetRenderTargets(_fiveRenderTargets)`.

I can open a DX12-specific issue if you would prefer that.
